ansible_collections/arista/avd/plugins/filter/convert_dicts.py

```diff
@@ -43,38 +42,41 @@ def convert_dicts(dictionary, primary_key="name", secondary_key="items"):
     Parameters
     ----------
     dictionary : any
-        Nested Dictionary to convert - returned untouched if not a nested dictionary
+        Nested Dictionary to convert - returned untouched if not a nested dictionary and list
     primary_key : str, optional
         Name of primary key used when inserting outer dictionary keys into items.
     secondary_key : str, optional
-        Name of secondary key used when inserting dictionary values which are list of dictionaries into items.
+        Name of secondary key used when inserting dictionary values which are list into items.
 
     Returns
     -------
     any
-        Returns list of dictionaries or input variable untouched if not a nested dictionary/list of dictionaries
+        Returns list of dictionaries or input variable untouched if not a nested dictionary/list.
     """
-    if not isinstance(dictionary, dict) or os.environ.get('AVD_DISABLE_CONVERT_DICTS'):
+    if not isinstance(dictionary, dict) and not isinstance(dictionary, list) or os.environ.get('AVD_DISABLE_CONVERT_DICTS'):
         # Not a dictionary, return the original
         return dictionary
+    elif isinstance(dictionary, list):
```

Comment:
we should not touch the input, if the list items are dicts. Only if they are something else.

Reply:
updated!